core/common/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/core/common/markdown/PostArtemisMarkdownTransformer.kt

```diff
@@ -1,26 +1,41 @@
 package de.tum.informatics.www1.artemis.native_app.core.common.markdown
 
+import de.tum.informatics.www1.artemis.native_app.core.common.R
+
 class PostArtemisMarkdownTransformer(val serverUrl: String, val courseId: Long) : ArtemisMarkdownTransformer() {
-    override fun transformExerciseMarkdown(title: String, url: String): String {
-        return "[$title](artemis:/$url)"
+
+    private val resourcePath = "android.resource://de.tum.cit.aet.artemis/"
+
+    override fun transformExerciseMarkdown(title: String, url: String, type: String): String {
+        val typeIcon =  when (type) {
+            "text" -> R.drawable.font_link_icon
+            "quiz" -> R.drawable.check_double_link_icon
+            "lecture" -> R.drawable.chalkboard_teacher_link_icon
+            "modeling" -> R.drawable.diagram_project_link_icon
+            "file-upload" -> R.drawable.file_arrow_up_link_icon
+            "programming" -> R.drawable.keyboard_link_icon
+            else -> return "[$title](artemis:/$url)"
+        }
+        return "![]($resourcePath$typeIcon)  [$title](artemis:/$url)"
     }
 
     override fun transformUserMentionMarkdown(text: String, fullName: String, userName: String): String = "[@$fullName](artemis://courses/$courseId/messages?username=$userName)"
 
     override fun transformChannelMentionMarkdown(
         channelName: String,
         conversationId: Long
-    ): String = "[#$channelName](artemis://courses/$courseId/messages?conversationId=$conversationId)"
+    ): String = "![]($resourcePath${R.drawable.message_link_icon})  [#$channelName](artemis://courses/$courseId/messages?conversationId=$conversationId)"
 
     override fun transformLectureContentMarkdown(
         type: String,
         fileName: String,
         url: String
     ): String {
+        val fileIconImage = "![]($resourcePath${R.drawable.file_link_icon})"
```

Comment:
I'd argue to create a private function for this (maybe something like`createFileTypeIconMarkdown(drawableId: @ResString int)`) and use it in the other `transform` methods as well

Reply:
Sounds good!